libpromises/evalfunction.c

```diff
@@ -3431,6 +3431,13 @@ static FnCallResult FnCallMergeData(EvalContext *ctx, ARG_UNUSED const Policy *p
             return FnFailure();
         }
 
+        // Ignore json primitives, only merge containers
+        if (JsonGetElementType(json) != JSON_ELEMENT_TYPE_CONTAINER)
+        {
+            JsonDestroy(json);
```

Comment:
Shouldn't this be destroyed only `if (allocated)` ?

Regarding your question, I believe somebody else is more appropriate to answer:

> I'm not sure about the consequences of this change, particularly what is the expected behavior of mergedata on a JSON primitive.

@tzz what do you say? :-)

Finally, if this is the right way, I would like to see a VERBOSE message here indicating that mergedata skipped merging something.

Reply:
I think that if some of the data is not mergeable, the function should explain in an error message and fail, to be consistent with the rest of CFE and to avoid data loss. Skipping data elements can be very hard to diagnose and debug.

Reply:
Updated to fail on non mergeable arguments, it seems more appropriate indeed.